File: LATIHAN1DPBO2023/CPP/main.cpp
// import library dan file
#include <bits/stdc++.h>
// #include "Mhs.cpp"
#include "Crud.cpp"

//menggunakan standarad namespace
using namespace std;

int main()
{
    /* deklarasi variabel*/
    int i, n = 0;
    string name;
    string nim;
    string ps;
    string fakultas;

    Crud crud;//-> deklrasi class crud
    list<Mhs> llist;//-> deklrasri class dalam list
    Mhs data; //-> deklrasi class Mahasiswa

    cin >> n; //-> inputan banyak nya data yg diinputkan
    for (i = 0; i < n; i++)
    {

        cin >> name >> nim >> ps >> fakultas;//-> input nama, nim, program studi, fakultras
        /* mengubah data sesuai inputan */
        data.set_name(name);
        data.set_nim(nim);
        data.set_ps(ps);
        data.set_fakultas(fakultas);

        crud.input(data, &llist);
    }

    /*nge print list mahasiswa*/
    cout << "List Mahasiswa : " << '\n';
    i = 0;
    for(list<Mhs>::iterator it = llist.begin(); it != llist.end(); it++)
    {
        cout << (i + 1) << ". " << it->get_name() << " " << it->get_nim() << " " << it->get_ps() << " " << it->get_fakultas() << '\n';
        i++;
    }

    int x;//-> variabel untuk milih ubah atau hapus
    cout << "Menu : "<< '\n';
    cout << "1. Ubah"<< '\n';
    cout << "2. Hapus"<< '\n';
    cin >> x;//-> input x

    if (x == 1)//-> jika 1 maka ubah
    {
        cout << "masukan nim yang akan diubah : " << '\n';
        cin >> nim;
        crud.ubah(nim, &llist);
    }
    else if(x == 2)//-> jika 2 maka hapus
    {
        cout << "masukan nim yang akan dihapus : " << '\n';
        cin >> nim;
        crud.hapus(nim, &llist);
    }
    /*nge print list mahasiswa*/
    cout << "List Mahasiswa : " << '\n';
    i = 0;
    for(list<Mhs>::iterator it = llist.begin(); it != llist.end(); it++)
    {
        cout << (i + 1) << ". " << it->get_name() << " " << it->get_nim() << " " << it->get_ps() << " " << it->get_fakultas() << '\n';
        i++;
    }
    return 0;
}

File: LATIHAN1DPBO2023/PYTHON/main.py
# import class file
from Mhs import Mhs
from Crud import Crud

masukan = [] # membuat variabel masukan menggunakan array

# iterasi
n = int(input())
for i in range(n):
    name = str(input())
    nim = str(input())
    ps = str(input())
    fakultas = str(input())
    
    masukan.append(Mhs(name, nim, ps, fakultas))

i = 0
for dataawal in masukan:
    print(str(i+1) + ".", dataawal.get_name(), "-", dataawal.get_nim(), "-", dataawal.get_ps(), "-", dataawal.get_fakultas())
    i += 1
    
nim = str(input())
Crud.ubah(nim, masukan)
i = 0
for dataubah in masukan:
    print(str(i+1) + ".", dataubah.get_name(), "-", dataubah.get_nim(), "-", dataubah.get_ps(), "-", dataubah.get_fakultas())
    i += 1
    
nim = str(input())
Crud.hapus(nim, masukan)
i = 0
for datahapus in masukan:
    print(str(i+1) + ".", datahapus.get_name(), "-", datahapus.get_nim(), "-", datahapus.get_ps(), "-", datahapus.get_fakultas())
    i += 1